src/main/java/pl/javastart/exercise/streams/FilteringMachine.java

```diff
@@ -1,23 +1,56 @@
 package pl.javastart.exercise.streams;
 
+import java.util.Collections;
+import java.util.Comparator;
 import java.util.List;
+import java.util.stream.Collectors;
 
 public class FilteringMachine {
 
     public List<Integer> filterOutNotEvenNumbers(List<Integer> numberList) {
-        return null;
+        List<Integer> notEvenNumbers = numberList.stream()
+                .filter(this::isNotEven)
+                .collect(Collectors.toList());
+        return notEvenNumbers;
+
     }
 
     public List<Integer> filterOutLowerNumbersThan20(List<Integer> numberList) {
-        return null;
+        List<Integer> lowerNumbersThan20 = numberList.stream()
+                .sorted()
+                .filter(this::isGreaterThen20)
+                .collect(Collectors.toList());
+        return lowerNumbersThan20;
     }
 
     public List<Book> convertToBooks(List<String> titles) {
-        return null;
+        List<Book> books = titles.stream()
+                .map(Book::new)
+                .collect(Collectors.toList());
+        return books;
     }
 
     // Metoda powinna przekształcić tytuły na książki i zwrócić tylko te które rozpoczynają się od słowa "Gra"
     public List<Book> convertToBooksAndReturnOnlyStartingWithGra(List<String> titles) {
-        return null;
+        List<Book> books = titles.stream()
+                .map(Book::new)
+//                .allMatch(book -> book.getTitle().contains("Gra"));
+                .filter(book -> book.getTitle().contains("Gra"))
```

Comment:
Tutaj lepiej by było najpierwsz przefiltrować tytuły a potem stworzyć książki za pomocą metody map. Jest to istotne bo nie zamienisz od razu wszystkich książek na obiekty, tylko te które rzeczywiście tymi książkami powinny być. Taka oszczędność pamięci troche :)